### PR TITLE
[10.0][BKP] l10n_it_fatturapa_out_rc: Backporting da 12.0

### PR DESCRIPTION
**Pull request by tafaRU**

### CODE REVIEW

File: l10n_it_fatturapa_out_rc/tests/test_e_invoice_out_rc.py [hunk 1]
```diff
@@ -46,7 +48,7 @@ def set_sequence_journal_selfinvoice(self, invoice_number, dt):
 
     def set_bill_sequence(self, invoice_number, dt):
         seq_pool = self.env['ir.sequence']
-        inv_seq = seq_pool.search([('name', '=', 'BILL Sequence')])[0]
+        inv_seq = seq_pool.search([('name', '=', 'Vendor Bills')])[0]
```
Comment (fredzamoabg) [inline l10n_it_fatturapa_out_rc/tests/test_e_invoice_out_rc.py:51]: non è più corretto utilizzare il parametro `limit` impostandolo ad `1`? metti caso che la search non ritorni nulla ti salterebbe un `index out of range`

**Review by fredzamoabg — CHANGES_REQUESTED** [review]

**Review by fredzamoabg — APPROVED** [review]

**Review by marco-marchiori — APPROVED** [review]
Functional test
- new fiscal position
- new RC type
- supplier bill
- self-invoice
- exported XML file
- checked XML file in SDI

test ok

**Comment by tafaRU** [comment]
Pare trattarsi di un errore temporaneo: https://app.travis-ci.com/github/OCA/l10n-italy/jobs/553087247#L254

/ocabot merge nobump

**Comment by tafaRU** [comment]
@sbidoul merge fails due to https://app.travis-ci.com/github/OCA/l10n-italy/jobs/553225335#L254

```
$ travis_install_nightly
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-PaO_Mh/lazy-object-proxy/
The command "travis_install_nightly" failed and exited with 1 during .
```

But the odd thing is that the other Travis builds don't fail in the same point, i.e: https://app.travis-ci.com/github/OCA/l10n-italy/jobs/553225336#L249

Could you please take a look?

Thank you in advance.

**Comment by tafaRU** [comment]
> It might be a transient error. I'd suggest retrying.

ok thanks @sbidoul, I thought same thing yesterday after https://github.com/OCA/l10n-italy/pull/2517#issuecomment-998836215
I retry and :crossed_fingers:

**Comment by sbidoul** [comment]
The problem might be due to https://github.com/ionelmc/python-lazy-object-proxy/issues/61. Since the travis build uses a very old pip version that does not support yanked releases, it may get an unsupported version of lazy-object-proxy. But it is still weird that it works in some builds.

**Comment by tafaRU** [comment]
It still fails: https://app.travis-ci.com/github/OCA/l10n-italy/jobs/553239081#L254

**Comment by SimoRubi** [comment]
> The problem might be due to [ionelmc/python-lazy-object-proxy#61](https://github.com/ionelmc/python-lazy-object-proxy/issues/61). Since the travis build uses a very old pip version that does not support yanked releases, it may get an unsupported version of lazy-object-proxy. But it is still weird that it works in some builds.

Why not updating Travis's `pip`? The old version already caused another issue (https://github.com/OCA/maintainer-quality-tools/issues/696) months ago

**Comment by sbidoul** [comment]
We can upgrade pip MQT...

Or use GitHub actions in this branch 10.0 too. I enabled runboat on l10n-italy 10.0 and triggered a [build](https://runboat.odoo-community.org/builds/b84ff09e5-2d86-4e51-9299-66985d00e1c8) for this PR so you can see how it goes. If runboat starts, there are great chances that the GitHub action will work too as it is based on the same image.

**Comment by tafaRU** [comment]
@sbidoul runboat started but failed due to the following error:

`ERROR: ./setup/l10n_it_fatturapa_out_rc is not a valid editable requirement. It should either be a path to a local project or a VCS URL (beginning with svn+, git+, hg+, or bzr+).`

**Comment by sbidoul** [comment]
That is because the setup directory is missing. The oca/oca-ci image now [generates](https://github.com/OCA/oca-ci/commit/c444546982560989716926503aafa5ac74c30750) it for you. So if you do a push on this PR it will rebuild and it should work.

**Comment by tafaRU** [comment]
@sbidoul with the rebuild we have:

- Travis fails with the same error raised during the attempt of merging by ocabot: https://app.travis-ci.com/github/OCA/l10n-italy/jobs/553579156#L267
- runboat is not shown

**Comment by sbidoul** [comment]
Runboat is now green (there was a little problem related to old pip versions with python 2). This means the GitHub actions tests should work too.

**Comment by tafaRU** [comment]
> This means the GitHub actions tests should work too.

Cool! What are the next steps?

**Comment by sbidoul** [comment]
> Cool! What are the next steps?

Well, there are several possibilities:

- Someone can debug why MQT travis_install_nightly does not work (but we'll switch off travis sooner or later)
- Someone can manually setup '.github/workflows` in this repo, by copying and adapting what is in the 14.0 or 15.0 branch (and perhaps keep travis with the lint mode only)
- Someone can add support for 10.0 (and 11, and 12) to oca/oca-addons-repo-template so this could be applied to other repos when we unplug travis. The hardest part (but not *that* hard) is to create a pre-commit config that is compatible with what MQT does today (i.e. flake 8 + pylint-odoo with the right config, and nothing else)

**Comment by tafaRU** [comment]
@sbidoul I'm trying to follow the following:

> * Someone can manually setup `.github/workflows` in this repo for 10.0, by copying and adapting what is in the 14.0 or 15.0 branch (and perhaps keep travis with the lint mode only)

You can see the current state here: https://github.com/tafaRU/l10n-italy/commits/10.0-add-gh_actions Test are OK!
But in light of the fact that in https://app.travis-ci.com/github/OCA/l10n-italy/builds/243860918 the only one build failed is that one related to *linting*, I'm not clear how to interpret the following part:

> (and perhaps keep travis with the lint mode only)

Could you please provide some guidance / advice on how to continue?

Thank you in advance.

**Comment by sbidoul** [comment]
@tafaRU cool.

What I'd do for linting is sketched in https://github.com/OCA/oca-addons-repo-template/issues/103.
Basically, create a `.pre-commit-config.yaml` with only the flake8 and pylint sections and  `.flake8` and `.pylinrc/.pylintrc-mandatory` that are inspired by [oca/maintainer-quality-tools](https://github.com/OCA/maintainer-quality-tools/tree/master/travis/cfg).

This way we'll have the same linter checks as in travis, but in pre-commit style.

**Comment by tafaRU** [comment]
@sbidoul do you see a problem simply upgrading pip as done in https://github.com/OCA/l10n-italy/pull/2517/commits/ab769210a51a3d4e996bd2b6ccf84a55c783f4b1 ?

**Comment by sbidoul** [comment]
If it's green it should be fine :)

**Pull request merged**